src/main/java/com/rocketden/main/controller/v1/RoomController.java

```diff
@@ -67,4 +69,9 @@ public RoomController(RoomService service) {
                                                       @RequestBody UpdateSettingsRequest request) {
         return new ResponseEntity<>(service.updateRoomSettings(roomId, request), HttpStatus.OK);
     }
+
+    @PostMapping("/rooms/{roomId}/spectator/{isSpectator}")
+    public ResponseEntity<RoomDto> setSpectator(@PathVariable String roomId, @PathVariable Boolean isSpectator, @RequestBody SetSpectatorRequest request) {
+        return new ResponseEntity<>(service.setSpectator(roomId, isSpectator, request), HttpStatus.OK);
+    }
```

Comment:
Thinking more, I'm somewhat split on how this current PR is designed, with regards to taking in room details. It seems that the request isn't actually using any of the room details, which calls into question the relevance of `roomId` and why this request is in the `RoomController.java` class and whatnot.

Also, this is unrelated to this PR, but I just realized that I'm not sure that we handle the case where a user is already in a room and tries to join another. Details like this will become especially relevant when we add accounts, and users could log in and thus realistically attempt to join two different rooms at the same time.

@alankbi, what are your thoughts on both of the above?

Reply:
Ah, I realized that you **are** using the room - you're allowing the host to set the user as a spectator. I think that actually makes sense, I'm not certain what this use case would be but perhaps that extra room management could be useful.

You can ignore the concerns about the room details! I think this structure makes sense 👍